pkg/recipes/terraform/config/config.go

```diff
@@ -150,13 +167,24 @@ func getProviderConfigs(ctx context.Context, requiredProviders []string, support
 			return nil, err
 		}
 		if len(config) > 0 {
-			providerConfigs[provider] = config
+			providerConfigs[provider] = []any{config}
 		}
 	}
 
 	return providerConfigs, nil
 }
 
+// getRecipeProviderConfigs returns the Terraform provider configurations for Terraform providers.
+func getRecipeProviderConfigs(ctx context.Context, envConfig *recipes.Configuration) map[string][]any {
+	// If the provider is not configured, return an empty config
+	if envConfig.RecipeConfig.Terraform.Providers.ProviderConfigs == nil {
```

Comment:
What does it do if envConfig.RecipeConfig is nil?

Reply:
code updated after rebase and tests added for nil checks